packages/graphile-utils/__tests__/makeAddPgTableOrderByPlugin.test.js

```diff
@@ -0,0 +1,355 @@
+import pg from "pg";
+import { graphql } from "graphql";
+import { createPostGraphileSchema } from "postgraphile-core";
+import { makeAddPgTableOrderByPlugin, orderByAscDesc } from "../node8plus";
+
+const clean = data => {
+  if (Array.isArray(data)) {
+    return data.map(clean);
+  } else if (data && typeof data === "object") {
+    return Object.keys(data).reduce((memo, key) => {
+      const value = data[key];
+      if (key === "id" && typeof value === "number") {
+        memo[key] = "[id]";
+      } else if (key === "nodeId" && typeof value === "string") {
+        memo[key] = "[nodeId]";
+      } else {
+        memo[key] = clean(value);
+      }
+      return memo;
+    }, {});
+  } else {
+    return data;
+  }
+};
+
+let pgPool;
+
+beforeAll(() => {
+  pgPool = new pg.Pool({
+    connectionString: process.env.TEST_DATABASE_URL,
+  });
+});
+
+afterAll(() => {
+  if (pgPool) {
+    pgPool.end();
+    pgPool = null;
+  }
+});
+
+const makePetsPlugin = nullsSortMethod =>
+  makeAddPgTableOrderByPlugin("graphile_utils", "users", build => {
+    const { pgSql: sql } = build;
+    const sqlIdentifier = sql.identifier(Symbol("pet"));
+
+    const customOrderBy = orderByAscDesc(
+      "PET_ID_AVERAGE", // this is a ridiculous and unrealistic column but it will serve for testing purposes
+      helpers => {
+        const { queryBuilder } = helpers;
+
+        const orderByFrag = sql.fragment`(
+          select avg(${sqlIdentifier}.id)
+          from graphile_utils.pets as ${sqlIdentifier}
+          where ${sqlIdentifier}.user_id = ${queryBuilder.getTableAlias()}.id
+        )`;
+
+        return orderByFrag;
+      },
+      { nulls: nullsSortMethod }
+    );
+
+    return customOrderBy;
+  });
+
+const getResultingOrderFromUserNodes = userNodes =>
+  userNodes.map(node => node.name);
+
+const checkArraysAreEqual = (array1, array2) =>
+  JSON.stringify(array1) === JSON.stringify(array2);
+
+const getSchemaAndPgClient = async nullsSortMethod => {
+  const schema = await createPostGraphileSchema(pgPool, ["graphile_utils"], {
+    disableDefaultMutations: true,
+    simpleCollections: "both",
+    appendPlugins: [makePetsPlugin(nullsSortMethod)],
+  });
+  const pgClient = await pgPool.connect();
```

Comment:
The `connect` and `release` should be paired, either using the callback pattern or by moving the connection into the tests themselves.